cmake/config_generation.cmake

```diff
@@ -54,65 +53,75 @@ set(_tbbbind_bin_version ${tbb_gen_cfg_TBBBIND_BINARY_VERSION})
-        set(TBB_IMPLIB_DEBUG "")
-        if (tbb_gen_cfg_HANDLE_SUBDIRS)
-            set(TBB_HANDLE_SUBDIRS "
-if (CMAKE_SIZEOF_VOID_P STREQUAL \"8\")
-    set(_tbb_subdir intel64/gcc4.8)
-else ()
-    set(_tbb_subdir ia32/gcc4.8)
-endif()
+
+        set (TBB_HANDLE_IMPLIB "
+            set (_tbb_release_dll \${_tbb_release_lib})
+            set (_tbb_debug_dll \${_tbb_debug_lib})
 ")
+        if (tbb_gen_cfg_HANDLE_SUBDIRS)
+            set(TBB_HANDLE_SUBDIRS "set(_tbb_subdir gcc4.8)")
 
             set(_libdir_for_pc_file "\${prefix}/lib/intel64/gcc4.8")
+            set(_tbb_pc_extra_libdir "-L\${prefix}/lib")
             configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb.pc @ONLY)
 
             set(_libdir_for_pc_file "\${prefix}/lib/ia32/gcc4.8")
+            set(_tbb_pc_extra_libdir "-L\${prefix}/lib32")
             configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb32.pc @ONLY)
         endif()
     elseif (tbb_gen_cfg_SYSTEM_NAME STREQUAL "Darwin")
         set(TBB_LIB_PREFIX "lib")
         set(TBB_LIB_EXT "\${_\${_tbb_component}_bin_version}.dylib")
-        set(TBB_IMPLIB_RELEASE "")
-        set(TBB_IMPLIB_DEBUG "")
+
+        set (TBB_HANDLE_IMPLIB "
+            set (_tbb_release_dll \${_tbb_release_lib})
+            set (_tbb_debug_dll \${_tbb_debug_lib})
+")
         set(_libdir_for_pc_file "\${prefix}/lib")
         configure_file(${_tbb_gen_cfg_path}/../integration/pkg-config/tbb.pc.in ${config_install_dir}/tbb.pc @ONLY)
     elseif (tbb_gen_cfg_SYSTEM_NAME STREQUAL "Windows")
         set(TBB_LIB_PREFIX "")
-        set(TBB_LIB_EXT "dll")
+        set(TBB_LIB_EXT "lib")
         set(TBB_COMPILE_DEFINITIONS "
                                   INTERFACE_COMPILE_DEFINITIONS \"__TBB_NO_IMPLICIT_LINKAGE=1\"")
-
-        # .lib files installed to TBB_LIB_REL_PATH (e.g. <prefix>/lib);
-        # .dll files installed to TBB_DLL_REL_PATH (e.g. <prefix>/bin);
-        # Expand TBB_LIB_REL_PATH here in IMPORTED_IMPLIB property and
-        # redefine it with TBB_DLL_REL_PATH value to properly fill IMPORTED_LOCATION property in TBBConfig.cmake.in template.
-        set(TBB_IMPLIB_RELEASE "
-                                      IMPORTED_IMPLIB_RELEASE \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}.lib\"")
-        set(TBB_IMPLIB_DEBUG "
-                                      IMPORTED_IMPLIB_DEBUG \"\${_tbb_root}/${TBB_LIB_REL_PATH}/\${_tbb_subdir}/\${_tbb_component}\${_bin_version}_debug.lib\"")
-        set(TBB_LIB_REL_PATH ${TBB_DLL_REL_PATH})
+        
+        # .lib - installed to TBB_LIB_REL_PATH (e.g. <prefix>/lib) and are passed as IMPORTED_IMPLIB_<CONFIG> property to target
+        # .dll - installed to <prefix>/bin or <prefix>/redist and are passed as IMPORTED_LOCATION_<CONFIG> property to target
+        set (TBB_HANDLE_IMPLIB "
+            find_file(_tbb_release_dll
```

Comment:
Do we need  a check if (EXISTS "${_tbb_release_lib}") here similar to line 97 ?

Reply:
It placed on line 97 because there is possibility to load only tbb release libraries via setting `TBB_FIND_RELEASE_ONLY` to 1. But release libraries must always be found.